src/components/splitButton/splitButton.ts

```diff
@@ -0,0 +1,400 @@
+import { LitElement, html } from 'lit';
+import {
+  customElement,
+  property,
+  state,
+  query,
+  queryAssignedElements,
+} from 'lit/decorators.js';
+import '../icon';
+
+import { classMap } from 'lit/directives/class-map.js';
+import downIcon from '@carbon/icons/es/chevron--down/24';
+
+import {
+  SPLIT_BTN_KINDS,
+  SPLIT_BTN_SIZES,
+  SPLIIT_BTN_ICON_POSITION,
+} from './defs';
+import './splitButtonOption';
+import SplitButtonScss from './splitButton.scss';
+import { ifDefined } from 'lit/directives/if-defined.js';
+
+/**
+ * Split Button
+ * @slot unnamed - Slot for split button options.
+ * @slot icon - Slot for an icon (optional).
+ * @fires on-change - Captures the event and emits the selected value and original event details.
+ */
+
+@customElement('kd-split-btn')
+export class SplitButton extends LitElement {
+  static override styles = SplitButtonScss;
+
+  /** @ignore */
+  static override shadowRootOptions = {
```

Comment:
I don't see `formAssociated` or `internals` like the core button component has. I think it needs both of those as well as a `type` attribute. This could be problematic, because those features might only apply to the `<button>`, and not to the secondary options in the dropdown menu. Basically, all of the options need to be able to trigger a submit on the parent `<form>` potentially. Will need to think carefully about how this all should work.

Reply:
Is split button going to be part of form ? As IMO it will used as inline element of table or some block level elements. But anyway will apply these on split button too. Thanks for clearing clouds. :)

Reply:
I asked @kaanhsu (Robert) and as per his opinion,

Split button will not be inside a form. The only place where we think it might be part of the form is, if instead the user prefer to save the form as a draft instead of submitting it right away